engine/equations.cc

```diff
@@ -760,7 +760,29 @@ namespace MathDAG
           }
       }
     return r;
-  }     
+  }
+  
+  VariableDAGPtr SystemOfEquations::getNodeFromVar(const VariablePtr v)
+  {
+    NodePtr r;
+    if (expressionCache.exists(*v))
+      return dynamic_pointer_cast<VariableDAG>(expressionCache[*v]);
+    else if (v && v->type()!=VariableBase::undefined) r=makeDAG(*v);
+    return dynamic_pointer_cast<VariableDAG>(r);
+  }         
+
+  ostringstream SystemOfEquations::getDefFromIntVar(const VariablePtr v)
+  {
+    ostringstream o;
+    
+	std::vector<VariableDAG*>::const_iterator it1=std::find_if(integrationVariables.begin(),integrationVariables.end(),[&](VariableDAG* i){return i->valueId==v->valueId();});    
+    if (*it1)     
```

Comment:
This bit of code is unnecessary. v already has a valueId() method, and getIntegralInput will return a null pointer if nothing has been cached. It won't mutate expressionCache's state.